hail/src/main/scala/is/hail/services/Requester.scala

```diff
@@ -19,7 +23,52 @@ import org.apache.log4j.{LogManager, Logger}
 import org.json4s.{DefaultFormats, Formats, JObject, JValue}
 import org.json4s.jackson.JsonMethods
 
+import scala.collection.JavaConverters._
 import scala.util.Random
+import java.io.FileInputStream
+
+
+abstract class CloudCredentials {
+  def accessToken(): String
+}
+
+class GoogleCloudCredentials(gsaKeyPath: String) extends CloudCredentials {
```

Comment:
I like how direct this whole file is, it feels like only the necessary core is written.